DOC-3065: Update object XML schema

rachelbritton opened the pull request:
Updated object XML schema on the Change Events page in the Admin Guide.

rachelbritton committed:
DOC-3065: Update object XML schema

Updated object XML schema on the Change Events page in the Admin Guide.

agudgeirsson commented on guides/admin-manual/admin_change_events.rst:
```diff
@@ -228,14 +241,86 @@ An example XML structure for a zone change script might look as follows for a zo
+        <type>13</type>
+        <server>dns1.corp.net.</server>
+        <view />
+        <zone>armann.test.</zone>
```
We should remove all references to `armann` from the examples

agudgeirsson commented on guides/admin-manual/admin_change_events.rst:
```diff
@@ -48,123 +48,141 @@ Script Interfaces
-Upon completion, the script must create a new XML structure and return it to Micetro Central. The schema for the XML structure that is returned is as follows:
-
-.. code-block:: XML
+Change Request Script Interface
```
Actually, change requests have a completely different structure. We should remove that part from this page (for now, until we extract that `.xml` file and create the corresponding `.xsd` schema)

agudgeirsson commented on guides/admin-manual/admin_change_events.rst:
```diff
@@ -228,14 +241,86 @@ An example XML structure for a zone change script might look as follows for a zo
+
+.. code-block:: XML
+
+  <xs:schema attributeFormDefault="unqualified" elementFormDefault="qualified" xmlns:xs="http://www.w3.org/2001/XMLSchema">
```
This is the same structure as in DNS zones so we can essentially remove this part and just mention that the structure above is both for DNS zones and records

rachelbritton commented on guides/admin-manual/admin_change_events.rst:
```diff
@@ -48,123 +48,141 @@ Script Interfaces
-  </customFields>
-  </result>
-
 Micetro Central uses the information in the XML structure to update other custom properties or to display an error message if the success attribute on the result element is set to 0. The following XML example shows how an error message can be returned by the change event script.
```
@agudgeirsson, is this returned error message example still valid? Or should that be removed as well?

rachelbritton committed:
DOC-3065. Added change request script.

Added the change request xml script.

agudgeirsson commented on guides/admin-manual/admin_change_events.rst:
```diff
@@ -6,236 +6,249 @@
   </xs:schema>
 
-An example XML structure with three custom properties named Location, Country and Region might look as follows:
+Micetro Central uses the information in the XML structure to either update other custom properties or display an error message if the success attribute on the result element is set to ``0``. The following XML example shows how an error message can be returned by the change event script:
```
This text was there before but it is a bit strange, for application return codes then 0 usually means success. This does not block me from accepting the pull-request but it is something to look into at some point

agudgeirsson commented on guides/admin-manual/admin_change_events.rst:
```diff
@@ -6,236 +6,249 @@
 
 .. code-block:: XML
+  
+  <xs:schema attributeFormDefault="unqualified" elementFormDefault="qualified" xmlns:xs="http://www.w3.org/2001/XMLSchema">
```
I don' think this block is needed, the schema defined above should be sufficient. I think an xml sample is only what is needed here

rachelbritton committed:
DOC-3065. Removed extra XML schema.

Removed the extra XML schema for DNS zone/record content change script, as requested.

rachelbritton committed:
DOC-3065. Revised structure.

Added section on Data Returned by Change Scripts.

agudgeirsson commented on guides/admin-manual/admin_change_events.rst:
```diff
@@ -6,236 +6,229 @@
+  </externalScriptParameters>
 
-.. _zone-content-change-script-interface:
+A zone content change script does not have any return value.
```
I think we can remove this, not sure if this sentence makes sense anymore given that we are only listing up XML structures at this point

rachelbritton committed:
DOC-3065. Removed return value statement.

Removed statement based on feedback.
